services/gitdiff/gitdiff.go

```diff
@@ -585,38 +592,49 @@ parsingLoop:
 				}
 				break parsingLoop
 			}
+
 			switch {
 			case strings.HasPrefix(line, cmdDiffHead):
 				break curFileLoop
 			case strings.HasPrefix(line, "old mode ") ||
 				strings.HasPrefix(line, "new mode "):
+
+				if strings.HasPrefix(line, "old mode ") {
+					curFile.OldMode = prepareValue(line, "old mode ")
+				}
+				if strings.HasPrefix(line, "new mode ") {
+					curFile.Mode = prepareValue(line, "new mode ")
+				}
+
 				if strings.HasSuffix(line, " 160000\n") {
 					curFile.IsSubmodule = true
 				}
+			case strings.HasPrefix(line, "new file mode "):
+				curFile.Mode = strings.TrimPrefix(line, "new file mode ")
```

Comment:
It seems that `NewMode` is a better field name? Its value always comes from "new mode " or "new file mode "

Reply:
I would leave it like that, because in other places we also use this naming principle.

a few lines above:
`file.OldName}}" data-new-filename="{{$file.Name`